models/attribute/src/types.ts

```diff
@@ -0,0 +1,23 @@
+import { Attribute, AttributeDefinition } from '@commercetools/platform-sdk';
+import type { TBuilder } from '@commercetools-test-data/core';
+
+// This is the REST representation
+export type TAttribute = Attribute;
+
+export type TReferenceExpandable = {
+  id: String;
+};
+
+// This is the generator "catchall" representation
+export type TAttributeDefault = TAttribute & {
+  attributeDefinition: AttributeDefinition;
+  referencedResource: TReferenceExpandable;
+  referencedResourceSet: Array<TReferenceExpandable>;
+};
```

Comment:
The generator is instead a combination of all the possible REST and graphql fields (minus `__typename`), and in the transformers, we can pick from among these

Reply:
💯